pkg/skaffold/build/tag/git_commit.go

```diff
@@ -89,9 +90,31 @@ func (c *GitCommit) GenerateFullyQualifiedImageName(workingDir string, imageName
 		return fmt.Sprintf("%s:%s-dirty", imageName, ref), nil
 	}
 
+	ref = gitTagToDockerTag(ref)
+
 	return fmt.Sprintf("%s:%s", imageName, ref), nil
 }
 
+func gitTagToDockerTag(text string) string {
+	withAuthorizedChars := regexp.MustCompile(`[^a-zA-Z0-9-._]`).ReplaceAllString(text, "_")
+
+	var authorizedPrefix string
+	for i := 0; i < len(withAuthorizedChars); i++ {
+		if withAuthorizedChars[i] == '-' || withAuthorizedChars[i] == '.' {
+			authorizedPrefix += "_"
+		} else {
+			break
+		}
+	}
+	withAuthorizedPrefix := authorizedPrefix + withAuthorizedChars[len(authorizedPrefix):]
+
+	if len(withAuthorizedPrefix) > 128 {
+		return withAuthorizedPrefix[0:128]
+	}
+
```

Comment:
I would also add a warning or maybe debug message for visibility - when tags start to overlap accidentally,this can be a useful hint